chain/client/src/client_actor.rs

```diff
@@ -1331,8 +1331,10 @@ impl ClientActor {
                     };
 
                 let me = self.client.validator_signer.as_ref().map(|x| x.validator_id().clone());
-                let shards_to_sync = (0..self.client.runtime_adapter.num_shards())
+                let epoch_id = self.client.chain.get_block_header(&sync_hash).unwrap().epoch_id();
+                let shards_to_sync = (0..self.client.runtime_adapter.num_shards(&epoch_id))
                     .filter(|x| {
+                        // TODO: this seems like it would break if epoch only has one epoch?
```

Comment:
Do you mean one block?

Reply:
Yes, I fixed this in https://github.com/near/nearcore/pull/4705